### PR TITLE
Feature/ntgl 880 group timetable view

### PR DESCRIPTION
**Pull request by jasttran**
- user context
- you can see what group clicked on.
- display group information for each group clicked on

**NOTE** to see functionality, uncomment line 33 in FriendsButton.tsx

image

### CODE REVIEW

File: client/src/components/sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -180,7 +181,7 @@ const Sidebar: React.FC = () => {
 
   return (
     <StyledSidebar ref={sideBarRef} collapsed={collapsed}>
-      {showGroupsSidebar && (
+      {!groupsSidebarCollapsed && (
```
Comment (Rayahhhmed) [inline client/src/components/sidebar/Sidebar.tsx:184]: good name change

File: server/src/user/user.service.ts [hunk 2]
```diff
@@ -348,16 +348,6 @@ export class UserService {
           },
         });
 
-        // convert members, groupAdmins and timetables objects into string ids
```
Comment (Rayahhhmed) [inline server/src/user/user.service.ts:351]: Good stuff removing all this

**Review by Rayahhhmed — APPROVED** [review]
Had a look at it, there are some console logs rest seems good to push

File: client/src/App.tsx [hunk 3]
```diff
@@ -602,8 +606,14 @@ const App: React.FC = () => {
...
+                  <>
+                    <TimetableTabs />
+                    <Timetable assignedColors={assignedColors} handleSelectClass={handleSelectClass} />
+                  </>
+                ) : (
+                  <TimetableShared assignedColors={assignedColors} handleSelectClass={handleSelectClass} />
+                )}
                 <ICSButton
                   onClick={() => downloadIcsFile(selectedCourses, createdEvents, selectedClasses, firstDayOfTerm)}
                 >
```
Comment (Shzmj) [inline client/src/App.tsx:619]: This is a probably a problem for future tickets that will deal with separating group timetable from individual user timetable: How do we handle the intermediate state when clicking on 'Shared Timetables' in sidebar (and have no friends - I'm guessing if you have a friend group then it displays the first group?):
image

**Review by Shzmj — APPROVED** [review]

File: client/src/App.tsx (hunk 3 above)
Comment (jasttran) [reply client/src/App.tsx:619]: good pick up. Lets make a new ticket for it and assign it to @Shzmj

**Review by Rayahhhmed — APPROVED** [review]

**Pull request merged**